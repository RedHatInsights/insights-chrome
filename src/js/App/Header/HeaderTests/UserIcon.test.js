import React from 'react';
import ConnectedUserIcon, { UserIcon }  from '../UserIcon';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallow, mount } from 'enzyme';

describe('Connected User Icon', () => {
    let initialState;
    let mockStore;
    beforeEach(() => {
        mockStore = configureStore();
        initialState = {
            chrome: {
                user: {
                    identity: {
                        user: {
                            username: 'test-user'
                        }
                    }
                }
            }
        };
    });

    it('should render correctly with initial state', () => {
        const store = mockStore(initialState);
        const wrapper = mount(
            <Provider store={store}>
                <ConnectedUserIcon/>
            </Provider>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe('User Icon', () => {

    it('should render correctly with initial state', () => {
        const account = {
            username: 'test'
        };
        const mockGetImage = jest.fn();
        const wrapper = shallow(
            <UserIcon account={account} getImage = { mockGetImage }/>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
