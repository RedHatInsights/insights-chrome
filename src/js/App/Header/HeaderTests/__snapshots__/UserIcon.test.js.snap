// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected User Icon should render correctly with initial state 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(UserIcon)>
    <UserIcon
      account={
        Object {
          "username": "test-user",
        }
      }
      dispatch={[Function]}
    >
      <Component
        alt="User Avatar"
        src="apps/chrome/assets/images/img_avatar.svg"
      >
        <img
          alt="User Avatar"
          className="pf-c-avatar"
          src="apps/chrome/assets/images/img_avatar.svg"
        />
      </Component>
    </UserIcon>
  </Connect(UserIcon)>
</Provider>
`;

exports[`User Icon should render correctly with initial state 1`] = `
<Component
  alt="User Avatar"
  src="apps/chrome/assets/images/img_avatar.svg"
/>
`;
